go/vt/vtgate/engine/route.go

```diff
@@ -455,7 +455,10 @@ func (route *Route) execInsertUnsharded(vcursor VCursor, queryConstruct *queryin
 	if err != nil {
 		return nil, fmt.Errorf("execInsertUnsharded: %v", err)
 	}
-	result.InsertID = uint64(insertid)
+
+	if insertid != 0 {
+		result.InsertID = uint64(insertid)
```

Comment:
nit: s/insertid/insertID/ because it's two words (and you use that spelling in the `result` struct as well).